lib/any.js

```diff
@@ -2,8 +2,20 @@
 
 module.exports = match
 
+var needsIndex = [
+  'first-child',
+  'first-of-type',
+  'last-child',
+  'last-of-type',
+  'nth-child',
+  'nth-last-child',
+  'nth-of-type',
+  'nth-last-of-type',
+  'only-child',
+  'only-of-type'
+]
+
 var zwitch = require('zwitch')
-var pseudo = require('./pseudo')
```

Comment:
this change can be reverted

Reply:
I am getting quite lost here :sad: 

What could be reverted, please?

Reply:
We were discussing two approaches.
1. place `needsIndex` in `any`, remove it from `pseudo`, which doesn’t work.
2. Pass `any` down

As 1 didn’t work, it can be reverted

Reply:
Pardon me, but I am not seeing why it doesn't work. Could you please point me to the right direction?

Reply:
You said [here in the issue](https://github.com/syntax-tree/unist-util-select/issues/9#issuecomment-744621815) that changing `needsIndex` had no effect

Reply:
I have made another commit https://github.com/syntax-tree/unist-util-select/pull/10/commits/89f87356ac213f685d5697446632c073f9e64a78 which I believe `needsIndex` into even better place. 

If there is need to revert something, it has to be by your reasoning as a reviewer, otherwise I suspect I wasn't clear somewhere in the issue and your interpretation of what I have stated is different from what I have meant.

Please, provide way of moving this PR further, discarding the "this change could be reverted" suggestions in the process, if possible.

Reply:
Also, I might start getting what do you mean. To make things clear further, BOTH "approaches" need to be applied, not just the second one (wit reverting the first one) to remove the cyclic dependency. 

It is true that I said the first one does not work, meaning it reduced the dependency from direct (A -> B -> A) to indirect (A -> B -> C -> A) which was further resolved by the approach 2.

Reply:
Or in even different words, there are TWO cyclic dependencies to begin with, the first approach removes the direct one (yellow), the second approach removes the indirect one (red)

![2020-12-15_11-50](https://user-images.githubusercontent.com/3952052/102205679-cdfb1700-3ecb-11eb-9227-74dccdc03c82.png)